### PR TITLE
feature/A2ZO-22: Add Fluent Validation

### PR DESCRIPTION
**Pull request by deba1**

### CODE REVIEW

File: src/Application/DTOs/EntityDTOs/ProductDTO.cs [hunk 1]
```diff
@@ -4,24 +4,13 @@ namespace Application.DTOs.EntityDTOs
 {
```
Comment (MahdiFaheem) [inline src/Application/DTOs/EntityDTOs/ProductDTO.cs:2]: Remove unused using statements.

File: src/Application/Managers/OrderManager.cs [hunk 2]
```diff
@@ -12,22 +15,56 @@ public interface IOrderManager : IBaseManager<Order>
...
         private readonly IOrderRepository _orderRepository;
+        private readonly IOrderItemRepository _orderItemRepository;
         private readonly IMapper _mapper;
         private readonly IBaseSecondLevelRepository<Order, Payment> _orderPaymentRepository;
+        private readonly ITransactionService _transactionService;
 
-        public OrderManager(IOrderRepository orderRepository, IMapper mapper, 
-            IBaseSecondLevelRepository<Order, Payment> orderPaymentRepository
+        public OrderManager(
+            IOrderRepository orderRepository,
```
Comment (MahdiFaheem) [inline src/Application/Managers/OrderManager.cs:30]: Fix object cycle.

File: src/Application/Validators/EntityDTOValidator/OrderValidator.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                        .Required()
+                        .MustAsync(async (productId, _) => (await productRepository.GetById(productId)) != null)
+                        .WithMessage("Product Doesn't exist.");
+
+                    orderItemRule.RuleFor(t => t.Quantity)
+                        .Required()
+                        .GreaterThan(0)
+                        .WithMessage("Quantity is required.");
+                });
+
```
Comment (MahdiFaheem) [inline src/Application/Validators/EntityDTOValidator/OrderValidator.cs]: Remove new line.

**Review by MahdiFaheem — CHANGES_REQUESTED** [review]

**Comment by MahdiFaheem** [comment]
FIx gender from string to enum in UserDTO

**Review by kanismim — COMMENTED** [review]
fix gender.

**Review by AJakif — APPROVED** [review]

**Review by MahdiFaheem — APPROVED** [review]

**Pull request merged**